frontend/simple/js/transactions.js

```diff
@@ -0,0 +1,263 @@
+      let transactionQueue = _transactionQueue.get(this)
+      let next = transactionQueue.pop()
+      while (next) {
+        // Establish a unique identifier for each transaction
+        if (!_transactionID.has(next)) {
+          let transactionID = _transactionID.get(this)
+          // Set transaction's ID and increment The next ID
+          _transactionID.set(next, transactionID)
+          _transactionID.set(this, transactionID + 1)
+          await transactionDB.setItem('nextID', _transactionID.get(this))
+        }
+        try {
+          await next.run()
+          let backup = await transactionDB.getItem(`${_transactionID.get(next)}`)
+          if (backup) {
+            let failures = _failures.get(this)
+            let index = failures.findIndex(failure => failure.id === _transactionID.get(next))
+            if (index > -1) {
+              failures.splice(index, 1)
+              await transactionDB.setItem('failures', failures)
+            }
+            await transactionDB.removeItem(`${_transactionID.get(next)}`)
+          }
+        } catch (ex) {
+          if (next.isRecoverable) {
+            // Save a serialized version of the transaction
+            let transactionID = _transactionID.get(next)
+            let failures = _failures.get(this)
+            let failure = failures.find(failure => failure.id === transactionID)
+            if (!failure) {
+              failure = { id: transactionID, retryCount: 0 }
+              failures.push(failure)
+              await transactionDB.setItem('failures', failures)
+            } else {
+              failure.retryCount++
+              await transactionDB.setItem('failures', failures)
+            }
+            await transactionDB.setItem(`${transactionID}`, next.toJSON())
+            let retry = next
+            setTimeout(() => {
+              this.run(retry)
+            }, 500 * fibSequence(failure.retryCount))
+          }
+        }
+        next = transactionQueue.pop()
+      }
+      _isRunning.set(this, false)
+    }
+  }
+}
+export let transactionQueue = new TransactionQueue()
+
+const api = {
+  '/v1/run': function (description, recoverable, steps) {
+    // If there are arguments then they want to run a specific transaction
+    // Otherwise start the queue
+    if (arguments.length) {
+      let transaction = new Transaction(description, recoverable)
+      for (let step of steps) {
+        // You may set variables with an array of names/value or a single name/value
+        if (step.execute === 'setInScope') {
+          for (let key of Object.keys(step.args)) {
+            console.log(key, step.args[key])
+            transaction.setInScope(key, step.args[key])
+          }
+        } else {
+          transaction.addStep(step)
+        }
+      }
+      transactionQueue.run(transaction)
+      return transaction.wait()
+    } else {
+      transactionQueue.run()
+    }
+  },
+  '/v1/enqueue': function (description, recoverable, steps) {
```

Comment:
This API is never used, and it's violation of DRY since it contains same code as above